src/Microsoft.ML.Data/DataView/DetectAnomalyBySrCnnBatchTransform.cs

```diff
@@ -0,0 +1,332 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.ML.Numeric;
+using Microsoft.ML.Runtime;
+
+namespace Microsoft.ML.Data.DataView
+{
+    public abstract class BatchTransformBase<TInput, TBatch> : IDataView
+    {
+        private protected sealed class Bindings : ColumnBindingsBase
+        {
+            private readonly DataViewType _outputColumnType;
+            private readonly int _inputColumnIndex;
+
+            public Bindings(DataViewSchema input, string inputColumnName, string outputColumnName, DataViewType outputColumnType)
+                : base(input, true, outputColumnName)
+            {
+                _outputColumnType = outputColumnType;
+                _inputColumnIndex = Input[inputColumnName].Index;
+            }
+
+            protected override DataViewType GetColumnTypeCore(int iinfo)
+            {
+                Contracts.Check(iinfo == 0);
+                return _outputColumnType;
+            }
+
+            // Get a predicate for the input columns.
+            public Func<int, bool> GetDependencies(Func<int, bool> predicate)
+            {
+                Contracts.AssertValue(predicate);
+
+                var active = new bool[Input.Count];
+                for (int col = 0; col < ColumnCount; col++)
+                {
+                    if (!predicate(col))
+                        continue;
+
+                    bool isSrc;
+                    int index = MapColumnIndex(out isSrc, col);
+                    if (isSrc)
+                        active[index] = true;
+                    else
+                        active[_inputColumnIndex] = true;
+                }
+
+                return col => 0 <= col && col < active.Length && active[col];
+            }
+        }
+
+        public bool CanShuffle => false;
+
+        public DataViewSchema Schema => SchemaBindings.AsSchema;
+
+        private readonly IDataView _source;
+        private readonly IHost _host;
+        private protected readonly Bindings SchemaBindings;
+        protected readonly string InputCol;
```

Comment:
>InputCol [](start = 34, length = 8)

This should also go in the deriving class (in order to support deriving classes that are not one-to-one).